Library/PTPusher.m

```diff
@@ -274,6 +274,20 @@ - (void)subscribeAll
   }
 }
 
+- (void)unsubscribeAllChannels
+{
+  if (channels && channels.count) {
+	NSMutableArray *channelsToUnsubscribe = [[NSMutableArray alloc] init];
+	  for (PTPusherChannel *channel in [channels allValues]) {
+		[channelsToUnsubscribe addObject:channel];
```

Comment:
The temporary array is unnecessary here. 'channels' is a dictionary, not an array, and you're already creating a new array of all the subscribed channels when you call 'allValues'. A crash would only occur if you were iterating over the dictionary itself.